src/apps/themes/crud.py

```diff
@@ -0,0 +1,97 @@
+from typing import Any
+
+from sqlalchemy import select
+from sqlalchemy.engine import Result
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm import Query
+
+from apps.themes.db.schemas import ThemeSchema
+from apps.themes.domain import PublicTheme, Theme, ThemeCreate, ThemeUpdate
+from apps.themes.errors import (
+    ThemeAlreadyExist,
+    ThemeNotFoundError,
+    ThemePermissionsError,
+    ThemesError,
+)
+from infrastructure.database.crud import BaseCRUD
+
+__all__ = ["ThemesCRUD"]
+
+
+class ThemesCRUD(BaseCRUD[ThemeSchema]):
+    schema_class = ThemeSchema
+
+    async def _fetch(self, key: str, value: Any) -> Theme:
+        """Fetch theme by id or display_name from the database."""
+
+        if key not in {"id"}:
+            raise ThemesError(
+                f"Can not make the looking up theme by {key} {value}"
+            )
```

Comment:
I think if you are iterating through any iterable object that excludes something or it is included to somewhere it should be moved out from the constraining you use it.

Example:
```python
ALLOWED_FIELDS = {"id"}
if key not in ALLOWED_FIELDS:
    ...
```